### PR TITLE
Implement arbitrary for up to size-12 tuples.

### LINKED ISSUE #1965: Provide `Arbitrary` implementation for tuples
Requested feature: It would be nice if Kani implements `Arbitrary` for tuples. 
Use case: To allow assigning symbolic constants to tuples, e.g.:
```rust
#[kani::proof]
fn main() {
    let x: (u8, i32) = kani::any();
}
```
instead of having to do:
```rust
    let x: (u8, i32) = (kani::any(), kani::any());
```
It perhaps makes sense to provide the implementation up to a certain tuple size, for example 12 similar to how `std` provides an implementation for `Debug` for tuples up to twelve items long:  https://doc.rust-lang.org/std/primitive.tuple.html#impl-Debug-for-(T%2C)

Link to relevant documentation (Rust reference, Nomicon, RFC):

### PR DESCRIPTION
**Pull request by YoshikiTakashima**
### Description of changes: 

Implements `kani::Arbitrary`  for tuples

### Resolved issues:

Resolves #1965 

### Related RFC:

Optional #ISSUE-NUMBER.

### Testing:

* How is this change tested? tests/kani/Intrinsics/SIMD/Operators/bitwise.rs

* Is this a refactor change? No

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [x] Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: library/kani/src/tuple.rs [hunk 1]
```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+//
+// Modifications Copyright Kani Contributors
```
Comment (zhassan-aws) [inline library/kani/src/tuple.rs:3]: Is there a reason this is using the "modifed-file" license header as opposed to the standard one?

File: library/kani/src/tuple.rs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+// See GitHub history for details.
+
+//! Support for arbitrary tuples where each element implements
+//! `kani::Arbitrary`. Tuples of size up to 11 are supported in this
```
Comment (zhassan-aws) [inline library/kani/src/tuple.rs:7]: Isn't this supporting up to 12 (A - L)?

File: library/kani/src/tuple.rs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+use crate::Arbitrary;
+
+/// This macro implements `kani::Arbitrary` on a tuple whose elements
+/// already implement `kani::Arbitrary` by running `kani::any()` on
+/// each index of the tuple.
+macro_rules! tuple {
+    ($($typ:ident),*) => {
+	impl<$($typ : Arbitrary),*>  Arbitrary for ($($typ,)*) {
+            #[inline(always)]
```
Comment (zhassan-aws) [inline library/kani/src/tuple.rs:18]: Is the indentation off on this line?

File: library/kani/src/tuple.rs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+	    fn any() -> Self {
+		($(crate::any::<$typ>(),)*)
```
Comment (zhassan-aws) [inline library/kani/src/tuple.rs:20]: The indentation seems off on this line as well. Perhaps because of tabs?

**Review by zhassan-aws — COMMENTED** [review]
Thanks @YoshikiTakashima! I have a few comments.

File: library/kani/src/tuple.rs (hunk 1 above)
Comment (YoshikiTakashima) [reply library/kani/src/tuple.rs:3]: No. I just copied the license header of whatever I had open next to it, which was likely the modified proptest. 

I'll change it back.

**Comment by YoshikiTakashima** [comment]
@zhassan-aws Thanks for the review. I'm a bit busy with other parts of kani, so I would get back to this PR around Jan. 2023. Have a nice holidays.

File: library/kani/src/tuple.rs (hunk 4 above)
Comment (YoshikiTakashima) [reply library/kani/src/tuple.rs]: 72a48f243631eef863686f1726db395a867df4a0

File: library/kani/src/tuple.rs (hunk 3 above)
Comment (YoshikiTakashima) [reply library/kani/src/tuple.rs:16]: 72a48f243631eef863686f1726db395a867df4a0

File: library/kani/src/tuple.rs (hunk 2 above)
Comment (YoshikiTakashima) [reply library/kani/src/tuple.rs]: 72a48f243631eef863686f1726db395a867df4a0, also fixed PR title so the squash commit says 12.

File: library/kani/src/tuple.rs (hunk 1 above)
Comment (YoshikiTakashima) [reply library/kani/src/tuple.rs]: 72a48f243631eef863686f1726db395a867df4a0

**Comment by zhassan-aws** [comment]
Thanks @YoshikiTakashima! Can we add a couple of tests for the minimum and maximum tuple sizes (i.e. 1 and 12) to make sure these are working correctly, and do not break in the future?

**Review by zhassan-aws — APPROVED** [review]
Thanks!
